Content.Shared/Body/Systems/SharedBodySystem.Organs.cs

```diff
@@ -0,0 +1,230 @@
+﻿using System.Diagnostics.CodeAnalysis;
+using Content.Shared.Body.Components;
+using Content.Shared.Body.Events;
+using Content.Shared.Body.Organ;
+using Content.Shared.Body.Part;
+using Content.Shared.Random.Helpers;
+using Robust.Shared.Containers;
+using Robust.Shared.GameStates;
+using Robust.Shared.Map;
+
+namespace Content.Shared.Body.Systems;
+
+public partial class SharedBodySystem
+{
+    private void InitializeOrgans()
+    {
+        SubscribeLocalEvent<OrganComponent, ComponentGetState>(OnOrganGetState);
+        SubscribeLocalEvent<OrganComponent, ComponentHandleState>(OnOrganHandleState);
+    }
+
+    private OrganSlot? CreateOrganSlot(string slotId, EntityUid parent, BodyPartComponent? part = null)
+    {
+        if (!Resolve(parent, ref part, false))
+            return null;
+
+        var slot = new OrganSlot(slotId, parent);
+        part.Organs.Add(slotId, slot);
+
+        return slot;
+    }
+
+    private bool CanInsertOrgan(EntityUid? organId, OrganSlot slot, OrganComponent? organ = null)
+    {
+        return organId != null &&
+               slot.Child == null &&
+               Resolve(organId.Value, ref organ, false) &&
+               _containers.TryGetContainer(slot.Parent, BodyContainerId, out var container) &&
+               container.CanInsert(organId.Value);
+    }
+
+    private void OnOrganGetState(EntityUid uid, OrganComponent organ, ref ComponentGetState args)
+    {
+        args.State = new OrganComponentState(organ.Body, organ.ParentSlot);
+    }
+
+    private void OnOrganHandleState(EntityUid uid, OrganComponent organ, ref ComponentHandleState args)
+    {
+        if (args.Current is not OrganComponentState state)
+            return;
+
+        organ.Body = state.Body;
+        organ.ParentSlot = state.Parent;
+    }
+
+    public bool InsertOrgan(EntityUid? organId, OrganSlot slot, OrganComponent? organ = null)
```

Comment:
organs have slots now? wasnt the idea that parts wouldnt have slots for organs and it would just fit however many up to a size limit? that makes a lot more sense to me

Reply:
like for example with slimes; they just have 2 organ slots for a core and lungs, so you couldn't ever retrofit a heart/stomach onto them in this system?

Reply:
You could if you create an organ slot on them, same as attaching new body parts, could use size instead though that system was never really used properly (partly because surgery never worked).
The size system isn't incompatible with this either, it's just to keep track of the organs and could be used to wrap a container slot too.

Reply:
yeah, i guess this is fine if its mainly for keeping track of organs